File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var min = prices[0];
        var currentMaxProfit = 0;
        for (var i = 1; i < prices.Length; i++)
        {
            if (prices[i] < min)
            {
                min = prices[i];
            }
            else
            {
                var profit = prices[i] - min;
                if (profit > currentMaxProfit)
                {
                    currentMaxProfit = profit;
                }
            }
        }
        
        return currentMaxProfit;
    }
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowest = prices[0]
        max_profit = 0

        for price in prices[1:]:
            profit = price - lowest
            if profit > max_profit: max_profit = profit
            elif price < lowest: lowest = price

        return max_profit
        